docs/multicast-guide.md

```diff
@@ -0,0 +1,153 @@
+`Multicast` must be enabled in the `antrea-controller` and `antrea-agent`
+configuration.
+
+```yaml
+  antrea-controller.conf: |
+    featureGates:
+    # Enable multicast traffic.
+      Multicast: true
+  antrea-agent.conf: |
+    # Enable multicast traffic.
+      Multicast: true
+    multicast:
+    # The names of the interfaces on Nodes that are used to forward multicast traffic.
+    # Defaults to transport interface if not set.
+      multicastInterfaces: 
+
+    # The interval at which the antrea-agent sends IGMP queries to Pods.
+    # Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+      igmpQueryInterval: "125s"
+```
+
+## Multicast NetworkPolicy
+
+Antrea NetworkPolicy and Antrea ClusterNetworkPolicy are supported for the following
+types of multicast traffic:
+
+1. IGMP egress rules: IGMP membership report and IGMP leave group messages.
+2. IGMP ingress rules: IGMP query, which includes IGMPv1, IGMPv2, and IGMPv3.
+3. Multicast egress rules: non-IGMP multicast traffic from Pod to other Pods or external hosts.
+Note: multicast ingress rules are currently not supported.
+
+Examples: [ACNP for IGMP traffic](antrea-network-policy.md#acnp-for-igmp-traffic) and
+[ACNP for multicast egress traffic](antrea-network-policy.md#acnp-for-multicast-egress-traffic).
+Please also check the meanings of special fields for Multicast NetworkPolicies from the paragraphs
+below these examples.
+
+## Debugging and collecting multicast statistics
+
+Users could check following guides and corresponding tools to get the current status of
+the multicast application running in the cluster.
+
+### Pod multicast group information
+
+This command prints the cluster-wide information of Pods that has joined multicast group
+throughout the whole cluster. Example:
+
+```bash
+kubectl get multicastgroups
+GROUP       PODS
+225.1.2.3   default/mcjoin, namespace/pod
+224.5.6.4   default/mcjoin
+```
+
+### Inbound and outbound multicast traffic statistics
+
+Please refer to the corresponding [antctl page](antctl.md#multicast-commands).
+
+### Multicast NetworkPolicy statistics
+
+[Antrea NetworkPolicystats](feature-gates.md#networkpolicystats) also apply to multicast NetworkPolicy.
+
+## Use case example
+
+In this section, we will show how multicast video steaming works in
+Antrea using [VLC](https://www.videolan.org/vlc/).
+
+To start a sample video streaming server, we create a Pod streaming sample video
+to the multicast IP address `239.255.12.42` with TTL 6.
+
+```bash
+kubectl run -i --tty --image=quay.io/galexrt/vlc:latest vlc-sender -- vlc --intf ncurses --vout dummy --aout dummy sample-video.mp4 --sout udp:239.255.12.42 --ttl 6 --repeat
+```
+
+It's expected to see outbound multicast traffic from this Pod
+by running `antctl get podmulticaststats` from Antrea Agent local to this Pod,
+which means the Pod is sending video streams outside the Pod.
+
+You will also find a multicast route, which indicates traffic from Antrea gateway
```

Comment:
You can also check the multicast routes on the Node by running command `ip mroute`, which should print the following route for forwarding the multicast traffic from the Antrea gateway interface to the transport interface